native-schema-registry/csharp/AWSGsrSerDe/AWSGsrSerDe.Tests/memory/AwsSchemaRegistryMemeoryLeakDetectionTests.cs

```diff
@@ -0,0 +1,83 @@
+using System.Collections.Generic;
+using AWSGsrSerDe.common;
+using AWSGsrSerDe.deserializer;
+using AWSGsrSerDe.serializer;
+using AWSGsrSerDe.serializer.json;
+using AWSGsrSerDe.Tests.utils;
+using Google.Protobuf;
+using NUnit.Framework;
+using static AWSGsrSerDe.Tests.utils.ProtobufGenerator;
+
+
+namespace AWSGsrSerDe.Tests.memory
+{
+    [TestFixture]
+    public class AwsSchemaRegistryMemeoryLeakDetectionTests
+    {
+        private static readonly Dictionary<string, dynamic> Configs = new Dictionary<string, dynamic>
+        {
+            { GlueSchemaRegistryConstants.AvroRecordType, AvroRecordType.GenericRecord },
+            { GlueSchemaRegistryConstants.DataFormatType, GlueSchemaRegistryConstants.DataFormat.AVRO },
+        };
+
+        private static readonly GlueSchemaRegistryKafkaSerializer KafkaSerializer = new GlueSchemaRegistryKafkaSerializer(Configs);
+        private static readonly GlueSchemaRegistryKafkaDeserializer KafkaDeserializer = new GlueSchemaRegistryKafkaDeserializer(Configs);
+
+        private void SerializeDeserializeProtobufMessage()
+        {
+            var message = (IMessage)BASIC_SYNTAX2_MESSAGE;
+            var configs = new Dictionary<string, dynamic>
+            {
+                { GlueSchemaRegistryConstants.ProtobufMessageDescriptor, message.Descriptor },
+                { GlueSchemaRegistryConstants.DataFormatType, GlueSchemaRegistryConstants.DataFormat.PROTOBUF },
+            };
+
+            KafkaSerializer.Configure(configs);
+            KafkaDeserializer.Configure(configs);
+
+            var serialized = KafkaSerializer.Serialize(message, message.Descriptor.FullName);
+
+            var deserializedObject =
+                KafkaDeserializer.Deserialize(message.Descriptor.FullName, serialized);
```

Comment:
We should call all the methods of serializer and deserializer to ensure 100% coverage. Add `getSchema` and `canDecode` methods.